pkg/crc/preflight/preflight_linux.go

```diff
@@ -146,21 +151,34 @@ func fixVsock() error {
 	if err != nil {
 		return err
 	}
-	_, _, err = crcos.RunWithPrivilege("modprobe vhost_vsock", "modprobe", "vhost_vsock")
+
+	udevRule := `KERNEL=="vsock", MODE="0060", OWNER="root", GROUP="libvirt"`
+
+	path, err := exec.LookPath("sudo")
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to find sudo on path: %w", err)
 	}
-	_, _, err = crcos.RunWithPrivilege("chown /dev/vsock", "chown", "root:libvirt", "/dev/vsock")
+	cmd := exec.Command(path, "tee", vsockUdevRulesPath)
+	cmd.Stdin = strings.NewReader(udevRule)
```

Comment:
not sure if a new method in `pkg/os/exec` for this would be better, something like `PrepareCmdWithPrivilege` that returns a `*exec.Cmd`?

Reply:
I think you can use `WriteToFileAsRoot` here?